src/main/java/org/takes/misc/Select.java

```diff
@@ -0,0 +1,133 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.NoSuchElementException;
+import java.util.Queue;
+
+/**
+ * Select elements into a new iterable with given condition.
+ *
+ * @author Jason Wong (super132j@yahoo.com)
+ * @version $Id$
+ * @since 0.13.8
+ */
+public final class Select<T> implements Iterable<T> {
+
+    /**
+     * Internal storage to hold the elements from iterables.
+     */
+    private final transient Iterable<T> list;
+
+    /**
+     * The condition to filter the element in the iterator.
+     */
+    private final transient Condition<T> condition;
+
+    /**
+     * To produce an iterable collection, determined by condition, combining a
+     * and b, with order of the elements in a first.
+     * @param itb Iterable to select
+     * @param cond To determine which element to add in the final iterable
+     */
+    public Select(final Iterable<T> itb, final Condition<T> cond) {
+        this.list = itb;
+        this.condition = cond;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new SelectIterator<T>(
+            this.list.iterator(),
+            this.condition
+        );
+    }
+
+    /**
+     * The select iterator to traverse the input iterables and return the
+     * elements from the list with given condition.
+     *
+     * <p>This class is NOT thread-safe.
+     */
+    private static class SelectIterator<E> implements Iterator<E> {
+
+        /**
+         * The iterator to reflect the traverse state.
+         */
+        private final transient Iterator<E> iterator;
+
+        /**
+         * The condition to filter the elements in the iterator.
+         */
+        private final transient Condition<E> condition;
+
+        /**
+         * The buffer storing the objects of the iterator.
+         */
+        private final transient Queue<E> buffer = new LinkedList<E>();
+
+        /**
+         * Ctor. ConcatIterator traverses the element.
+         * @param itr Iterator of the original iterable
+         * @param cond Condition to filter out elements
+         */
+        public SelectIterator(final Iterator<E> itr, final Condition<E> cond) {
+            this.condition = cond;
+            this.iterator = itr;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (this.buffer.isEmpty()) {
+                while (this.iterator.hasNext()) {
+                    final E object = this.iterator.next();
+                    if (this.condition.fits(object)) {
+                        buffer.add(object);
```

Comment:
there should be a `break` after this line. we don't want to fetch everything, just one element is enough